### ISSUE TITLE
ERR_CONNECTION_REFUSED on /auth/callback

### ISSUE THREAD

**Issue by Juggo** [pozil/salesforce-react-integration#1]
Hello,

I have created a Connected app inside my Salesforce, enabled OAuth and added a callback Uri that is identical to the one in the sample app. I have also replaced the consumer key and secret in the sample app for the ones in my SF Connected app, yet after entering and submitting my Salesforce login details in the React app, ERR_CONNECTION_REFUSED is being thrown on the http://localhost:3000/auth/callback. But according to Salesforce Login history I succesfully login.

Might it have something to do with CORS or an older TLS perhaps?

My config:

```
var config = {};


// Salesforce client settings for Force.com connection
config.sfdc = {
  // OAuth2 service
  auth : {
    // OAuth authentication domain
    // For production or DE use
    domain : 'https://login.salesforce.com',
    // For sandbox use
    //domain : 'https://test.salesforce.com',

    // URL called by Force.com after authorization and used to extract an authorization code.
    // This should point to your app and match the value configured in your App in SFDC setup)
    callbackUrl : 'http://localhost:3000/auth/callback',

    // Set of secret keys that allow your app to authenticate with Force.com
    // These values are retrieved from your App configuration in SFDC setup.
    // NEVER share them with a client.
    consumerKey : 'my key',
    consumerSecret : 'my secret',
  },
  // Data service
  data : {
    // Force.com API version
    apiVersion : 'v41.0'
  }
};


// Express server configuration
config.server = {
  // Server HTTP port
  port : 3000,

  // Whether the server is configured with HTTPS
  isHttps : false,

  // Secret key used to encrypt user sessions
  sessionSecretKey : 'mySecretK3y!'
};


module.exports = config;
```

**Comment by pozil** [comment]
Hi @Juggo,

I just re-deployed the app to try it out because it's has been some time I haven't run it.
It still works for me without any modification other than setting the consumer key and secret.

Could you share more details about the ERR_CONNECTION_REFUSED error?
I assume that it is reported by Node but is that the case?

If it is indeed Node, I need to ask if you modified `server.js`?

I don't think that CORS is the problem here but could you double check that the callback url uses the right port and is identical to the one configured in your Connected App?

Also, could this be due to a network problem on your machine (firewall, invalid config)?
The authentication will fail if the callback URL cannot be reached.

Cheers,

**Comment by Juggo** [comment]
![bez nazvu](https://user-images.githubusercontent.com/13488677/33952610-a5f02408-e032-11e7-94c3-161161878d2c.png)
(The code parameter was removed.)

The callback url is the same in my Connected app : http://localhost:3000/auth/callback. 
I am not sure but I do not think there is a problem on my machine, I managed to query from Salesforce on my localhost using this sample https://github.com/mmurph23/sf-connect-dev, although I am having problems deploying it to Heroku.

Could it be a Proxy issue?
